_dev/src/store/modules/product-feed/actions.ts

```diff
@@ -130,18 +130,33 @@ export default {
     const productFeedSettings = state.settings;
     const targetCountries = changeCountriesNamesToCodes(getters.GET_TARGET_COUNTRIES);
     const attributeMapping = getDataFromLocalStorage('productFeed-attributeMapping') || {};
+    const deliveryEnabled = productFeedSettings.deliveryDetails.filter((e) => e.enabledCarrier);
+    const shipping = productFeedSettings.shippingSettings;
     commit(MutationsTypes.SET_SELECTED_PRODUCT_FEED_SETTINGS, {
       name: 'attributeMapping', data: attributeMapping,
     });
+
+    deliveryEnabled.forEach((delivery, index) => {
+      if ((delivery.minHandlingTimeInDays === null && delivery.maxHandlingTimeInDays === null)
+      || (delivery.minTransitTimeInDays === null && delivery.maxTransitTimeInDays === null)) {
+        deliveryEnabled.splice(index);
+        shipping.forEach((e, i) => {
+          if (e.properties.id_carrier === delivery.carrierId) {
+            shipping.splice(i);
```

Comment:
Correct me if I'm right, but this will impact the variable `shipping`, that is a reference to the variable `productFeedSettings.shippingSettings;`, itself a reference to `state.settings`. With this line, you're altering the state.

Is there any risk of side effect? Especially if we run the product feed funnel several times without refreshing?

Reply:
Yes ur right about that, I will change it with a copy from `productFeedSettings.shippingSettings`